php/query_utils.php

```diff
@@ -462,6 +462,95 @@ function get_associated_survey_instances(
     return $ret;
 }
 
+/**
+ * Get a list of all surveys associated with a user_id. The
+ * search may be narrowed by further specifying a `course_code` and a `term`
+ */
+function get_associated_surveys(
+    $user_id,
+    $course_code = null,
+    $term = null,
+    $conn = null
+) {
+    // if we don't specify these parameters, make them
+    // SQL wildcards
+    if ($course_code == null) {
+        $course_code = "%";
+    }
+    if ($term == null) {
+        $term = "%";
+    }
+    $roles = [
+        "dept_survey_choices" => "dept_survey_choice_id",
+        "course_survey_choices" => "course_survey_choice_id",
+        "ta_survey_choices" => "ta_survey_choice_id"
+    ];
+    //check user_role
+    $sql =
+        "SELECT is_admin, is_instructor, is_ta FROM user_associations AS a " .
+        "JOIN users AS u " .
+        "ON u.user_id = a.user_id " .
+        "WHERE a.user_id = :user_id AND a.course_code LIKE :course_code";
+    $bound = [
+        "user_id" => $user_id,
+        "course_code" => $course_code
+    ];
+    $res = do_select_query($sql, $bound, $conn);
+    $table_name = [];
+    $user = $res->result[0];
+    //higher role takes precidence as of now
+    if ($user["is_admin"]) {
+        $table_name[] = "dept_survey_choices";
+    }
+    if ($user["is_instructor"]) {
+        $table_name[] = "course_survey_choices";
+    }
+    if ($user["is_ta"]) {
+        $table_name[] = "ta_survey_choices";
+    }
+    // unpack the query to be a regular list of ids
```

Comment:
Spell out what you're doing here. Explain the algorithm and how your queries relate to it.